src/testWithSpringBoot_3_0/java/org/openrewrite/java/spring/boot3/MigrateResponseEntityExceptionHandlerHttpStatusToHttpStatusCodeTest.java

```diff
@@ -0,0 +1,158 @@
+                      return super.handleExceptionInternal(ex, body, headers, status, request);
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void convertHttpUsageAsInt() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import org.springframework.http.HttpHeaders;
+              import org.springframework.http.HttpStatus;
+              import org.springframework.http.ResponseEntity;
+              import org.springframework.web.context.request.WebRequest;
+              import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+              
+              class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
+              
+                  @Override
+                  protected ResponseEntity<Object> handleExceptionInternal(Exception ex, Object body, HttpHeaders headers, HttpStatus status, WebRequest request) {
+                      int value = status.value();
+                      return super.handleExceptionInternal(ex, body, headers, status, request);
+                  }
+              }
+              """,
+            """
+              import org.springframework.http.HttpHeaders;
+              import org.springframework.http.HttpStatusCode;
+              import org.springframework.http.ResponseEntity;
+              import org.springframework.web.context.request.WebRequest;
+              import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+              
+              class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
+              
+                  @Override
+                  protected ResponseEntity<Object> handleExceptionInternal(Exception ex, Object body, HttpHeaders headers, HttpStatusCode status, WebRequest request) {
+                      int value = status.value();
+                      return super.handleExceptionInternal(ex, body, headers, status, request);
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void convertHttpUsageAsEnum() {
+        //language=java
+        rewriteRun(
+          java(
+            """
+              import org.springframework.http.HttpHeaders;
+              import org.springframework.http.HttpStatus;
+              import org.springframework.http.ResponseEntity;
+              import org.springframework.web.context.request.WebRequest;
+              import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+              
+              class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
+              
+                  @Override
+                  protected ResponseEntity<Object> handleExceptionInternal(Exception ex, Object body, HttpHeaders headers, HttpStatus status, WebRequest request) {
+                      HttpStatus enumValue = status.value();
```

Comment:
... whereas over here it returns an `HttpStatus`. That reads kind of weird to me; what's the difference between these two text blocks?

Reply:
This specific usage was invalid, I've changed the testcase to use HttpStatus enum as that implements HttpStatusCode and therefore is a valid type to return.